### ISSUE TITLE
Map moves (and reloads) when callout near border

### ISSUE THREAD

**Issue by mihaeu** [mihaeu/warmshowers-ios#9]
Whenever a pin near the map border is clicked, the callout pushes the map, forcing it to reload and closing the callout.

**Comment by mihaeu** [comment]
This is probably due to the way the removal and adding of the pins is handled.

See

``` Swift
func loadUserAnnotations(users: [Int:User])
    {
        // todo
        mapView.removeAnnotations(userAnnotations)
        userAnnotations.removeAll()

        for (id, user) in users {
            if user.longitude != 0.0 {
                userAnnotations.append(UserAnnotation(user: user))
            }
        }

        mapView.addAnnotations(userAnnotations)
    }
```

**Comment by mihaeu** [comment]
Fixed in 684f82b

Only partial reload works, but I still need to refactor the solution.
